app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt

```diff
@@ -54,12 +59,39 @@ class RoutePresenter(
             is RouteAction.OpenWebsite -> {
                 openWebsite(destination.url)
             }
+            is RouteAction.SystemSetting -> {
+                openSetting(destination.setting.settingIntent)
+            }
```

Comment:
1. Unpack `settingIntent` in the `openSetting` method, not here.
2. We should bite the bullet and rename `destination` to `action`.